src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -82,17 +82,15 @@ public static String removeUnwantedCharacters(String key, String unwantedCharact
         for (char ch : unwantedCharacters.toCharArray()) {
```

Comment:
We don't need the `unwantedChars` variable, do we? Can you please delete that for loop and the variable? --> the code on line 86 directly accesses the method parameter `unwantedChararacters`!

Side note: This is an example why similar variable names can be mixed up.

Reply:
Of course, we don't need it, I'm sorry. Every time I get the impression that I don't notice something. 🐞